gcovr/workers.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding:utf-8 -*-
+
+# This file is part of gcovr <http://gcovr.com/>.
+#
+# Copyright 2013-2018 the gcovr authors
+# Copyright 2013 Sandia Corporation
+# This software is distributed under the BSD license.
+
+
+from threading import Thread, Lock
+from contextlib import contextmanager
+
+import sys
+if sys.version_info[0] >= 3:
+    from queue import Queue, Empty
+else:
+    from Queue import Queue, Empty
+
+
+class LockedDirectories(object):
+    """
+    Class that keeps a list of locked directories
+    """
+    def __init__(self):
+        self.lock = Lock()
+        self.dirs = set()
+
+    def run_in(self, dir_):
```

Comment:
This shall probably be written using a while loop rather than using recursion.
This shall probably use a condition variable rather than just a lock to prevent the usage of `time.sleep`

Reply:
@mayeut Fair go! I've made the change and rebased again.

Reply:
:+1: